File: python/mlaa.py
"""Author:Manjuram G Prabhudessai
//Roll No :24
//Start date:08/07/2024
//Modified date:15/07/2024
/*Description:This program performs all the opration which happens in Goa legislative assembly.
It shows list of mla,cabinet ministers and also gives the information about them.
It also has the House session in which the user can add,remove,pas and display the bill """


class Bill:
    def __init__(self, title, description):
        self.title = title
        self.description = description
        self.is_passed = False

    def get_title(self):
        return self.title

    def pass_bill(self):
        self.is_passed = True

    def __str__(self):
        return f"Title: {self.title}\nDescription: {self.description}\nPassed: {'Yes' if self.is_passed else 'No'}\n"


class Mla:
    def __init__(self):
        self.bills = []

    def list_mla(self):
        while True:
            print("******Welcome to Legislature*********")
            print("1: The Speaker")
            print("2: Deputy Speaker")
            print("3: Opposition Leader")
            print("4: MLA's")
            print("5: Exit")
            lan = input("Please Enter your choice: ")

            if lan == "1":
                print("************The Speaker************")
                print("Name: Shri Ramesh Tawadkar")
                print("Email: ram@#$@gmail.com")
                print("Address: Office of speaker, Goa legislative Assembly\n")
            elif lan == "2":
                print("***********The Deputy Speaker*******************")
                print("Name: Shri. Joshua De Souza")
                print("Email: jous@email id")
                print("Address: Office of speaker, Goa legislative Assembly\n")
            elif lan == "3":
                print(
                    "***********************The Opposition Leader***********************"
                )
                print("Name: Shri. Yuri Alemao")
                print("Constituency No/Name: 34 - Cuncolim")
            elif lan == "4":
                print("***********************The MLA's***********************")
                print("Name: Digambar Kamat")
                print("Party: BJP\n")
                print("Name: SHRI. JIT AROLKAR")
                print("Place: Mandrem")
                print("Party: Maharashtrawadi Gomantak Party\n")
                print("Name: SHRI. NILKANTH HALARNKAR")
                print("Place: Tivim")
                print("Party: Bharatiya Janata Party\n")
                print("Name: SHRI. Subash SHirodkar")
                print("Place: Shiroda")
                print("Party: Bharatiya Janata Party\n")
            elif lan == "5":
                return
            else:
                print("Invalid choice")

    def add_bill(self):
        title = input("Enter bill title: ")
        description = input("Enter bill description: ")
        self.bills.append(Bill(title, description))

    def remove_bill(self):
        title = input("Enter title of bill to remove: ")
        self.bills = [
            bill for bill in self.bills if bill.get_title().lower() != title.lower()
        ]

    def pass_bill(self):
        title = input("Enter title of bill to pass: ")
        for bill in self.bills:
            if bill.get_title().lower() == title.lower():
                bill.pass_bill()
                print("Bill passed successfully.")
                return
        print("Bill not found.")

    def display_bills(self):
        print("***********************The Bills***********************")
        for bill in self.bills:
            print(bill)

    def manage_bills(self):
        while True:
            print("*********WELCOME TO THE BILL SESSION**********")
            print("Choose an option:")
            print("1. Add Bill")
            print("2. Remove Bill")
            print("3. Pass Bill")
            print("4. Display Bills")
            print("5. Exit")

            choice = input()
            if choice == "1":
                self.add_bill()
            elif choice == "2":
                self.remove_bill()
            elif choice == "3":
                self.pass_bill()
            elif choice == "4":
                self.display_bills()
            elif choice == "5":
                return
            else:
                print("Invalid option. Please try again.")

    def council(self):
        print("Welcome to the Council of Ministers")
        while True:
            print("1: Home Minister")
            print("2: Health Minister")
            print("3: PWD Minister")
            print("4: Finance Minister")
            print("5: IT Minister")
            men = input("PLEASE Enter your choice: ")

            if men == "1":
                print("************Home Minister************")
                print("Name: Dr Pramod Sawant\n")
            elif men == "2":
                print("***********Health Minister******************")
                print("Name: Shri Vishwajit Rane\n")
            elif men == "3":
                print("***********PWD Minister******************")
                print("Name: Shri Sudhin Dhavlikar\n")
            elif men == "4":
                print("***********Finance Minister******************")
                print("Name: Pramod Sawant\n")
            elif men == "5":
                return
            else:
                print("Invalid choice")

    def the_government(self):
        while True:
            print("Know your government")
            print("1: The Governor")
            print("2: Chief Minister")
            print("3: Council Of Ministers")
            print("4: Exit")
            mla = input("Enter your choice: ")

            if mla == "1":
                print("************The Governor************")
                print("Name: Shri. P. S. Sreedharan Pillai")
                print("Email: ram@#$@gmail.com")
                print("Address: Office of speaker, Goa legislative Assembly\n")
            elif mla == "2":
                print("***********The Chief Minister*******************")
                print("Name: Shri. Pramod Sawant")
                print("Email: jous@email id")
                print("Address: Office of speaker, Goa legislative Assembly\n")
            elif mla == "3":
                self.council()
            elif mla == "4":
                return
            else:
                print("Invalid choice")


class Assembly(Mla):
    def my_assemble(self):
        while True:
            print("1: The Legislature")
            print("2: The Government")
            print("3: The House")
            print("4: Contact Us")
            print("5: Exit")
            menu = input("Enter your choice: ")

            if menu == "1":
                print("The Legislature:")
                self.list_mla()
            elif menu == "2":
                print("The Government:")
                self.the_government()
            elif menu == "3":
                print("The House:")
                self.manage_bills()
            elif menu == "4":
                print("Contact Us:")
                print("Email: info@goa.gov.in\n")
            elif menu == "5":
                print("Exiting...")
                return
            else:
                print("Invalid choice")


if __name__ == "__main__":
    assembly = Assembly()
    assembly.my_assemble()

File: c#/mlaa.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Bill {
    public string Title { get; }
    public string Description { get; }
    public bool IsPassed { get; private set; }

    public Bill(string title, string description) {
        Title = title;
        Description = description;
        IsPassed = false;
    }

    public void Pass() {
        IsPassed = true;
    }

    public override string ToString() {
        return $"Title: {Title}\nDescription: {Description}\nPassed: {(IsPassed ? "Yes" : "No")}\n";
    }
}

class Mla {
    protected List<Bill> bills = new List<Bill>();

    public void ListMla() {
        while (true) {
            Console.WriteLine("******Welcome to Legislature*********");
            Console.WriteLine("1: The Speaker");
            Console.WriteLine("2: Deputy Speaker");
            Console.WriteLine("3: Opposition Leader");
            Console.WriteLine("4: MLA's");
            Console.WriteLine("5: Exit");
            Console.Write("Please Enter your choice: ");
            int lan = int.Parse(Console.ReadLine());

            switch (lan) {
                case 1:
                    Console.WriteLine("************The Speaker************");
                    Console.WriteLine("Name: Shri Ramesh Tawadkar");
                    Console.WriteLine("Email: ram@#$@gmail.com");
                    Console.WriteLine("Address: Office of speaker, Goa legislative Assembly\n");
                    break;
                case 2:
                    Console.WriteLine("***********The Deputy Speaker*******************");
                    Console.WriteLine("Name: Shri. Joshua De Souza");
                    Console.WriteLine("Email: jous@email id");
                    Console.WriteLine("Address: Office of speaker, Goa legislative Assembly\n");
                    break;
                case 3:
                    Console.WriteLine("***********************The Opposition Leader***********************");
                    Console.WriteLine("Name: Shri. Yuri Alemao");
                    Console.WriteLine("Constituency No/Name: 34 - Cuncolim");
                    break;
                case 4:
                    Console.WriteLine("***********************The MLA's***********************");
                    Console.WriteLine("Name: Digambar Kamat");
                    Console.WriteLine("Party: BJP\n");
                    Console.WriteLine("Name: SHRI. JIT AROLKAR");
                    Console.WriteLine("Place: Mandrem");
                    Console.WriteLine("Party: Maharashtrawadi Gomantak Party\n");
                    Console.WriteLine("Name: SHRI. NILKANTH HALARNKAR");
                    Console.WriteLine("Place: Tivim");
                    Console.WriteLine("Party: Bharatiya Janata Party\n");
                    Console.WriteLine("Name: SHRI. Subash SHirodkar");
                    Console.WriteLine("Place: Shiroda");
                    Console.WriteLine("Party: Bharatiya Janata Party\n");
                    break;
                case 5:
                    return;
                default:
                    Console.WriteLine("Invalid choice");
                    break;
            }
        }
    }

    public void AddBill() {
        Console.Write("Enter bill title: ");
        string title = Console.ReadLine();
        Console.Write("Enter bill description: ");
        string description = Console.ReadLine();
        bills.Add(new Bill(title, description));
    }

    public void RemoveBill() {
        Console.Write("Enter title of bill to remove: ");
        string title = Console.ReadLine();
        bills = bills.Where(bill => !bill.Title.Equals(title, StringComparison.OrdinalIgnoreCase)).ToList();
    }

    public void PassBill() {
        Console.Write("Enter title of bill to pass: ");
        string title = Console.ReadLine();
        Bill bill = bills.FirstOrDefault(b => b.Title.Equals(title, StringComparison.OrdinalIgnoreCase));
        if (bill != null) {
            bill.Pass();
            Console.WriteLine("Bill passed successfully.");
        } else {
            Console.WriteLine("Bill not found.");
        }
    }

    public void DisplayBills() {
        Console.WriteLine("***********************The Bills***********************");
        foreach (Bill bill in bills) {
            Console.WriteLine(bill);
        }
    }

    public void ManageBills() {
        while (true) {
            Console.WriteLine("*********WELCOME TO THE BILL SESSION**********");
            Console.WriteLine("Choose an option:");
            Console.WriteLine("1. Add Bill");
            Console.WriteLine("2. Remove Bill");
            Console.WriteLine("3. Pass Bill");
            Console.WriteLine("4. Display Bills");
            Console.WriteLine("5. Exit");
            int choice = int.Parse(Console.ReadLine());

            switch (choice) {
                case 1:
                    AddBill();
                    break;
                case 2:
                    RemoveBill();
                    break;
                case 3:
                    PassBill();
                    break;
                case 4:
                    DisplayBills();
                    break;
                case 5:
                    return;
                default:
                    Console.WriteLine("Invalid option. Please try again.");
                    break;
            }
        }
    }

    public void Council() {
        Console.WriteLine("Welcome to the Council of Ministers");
        while (true) {
            Console.WriteLine("1: Home Minister");
            Console.WriteLine("2: Health Minister");
            Console.WriteLine("3: PWD Minister");
            Console.WriteLine("4: Finance Minister");
            Console.WriteLine("5: IT Minister");
            Console.Write("PLEASE Enter your choice: ");
            int men = int.Parse(Console.ReadLine());

            switch (men) {
                case 1:
                    Console.WriteLine("************Home Minister************");
                    Console.WriteLine("Name: Dr Pramod Sawant\n");
                    break;
                case 2:
                    Console.WriteLine("***********Health Minister******************");
                    Console.WriteLine("Name: Shri Vishwajit Rane\n");
                    break;
                case 3:
                    Console.WriteLine("***********PWD Minister******************");
                    Console.WriteLine("Name: Shri Sudhin Dhavlikar\n");
                    break;
                case 4:
                    Console.WriteLine("***********Finance Minister******************");
                    Console.WriteLine("Name: Pramod Sawant\n");
                    break;
                case 5:
                    return;
                default:
                    Console.WriteLine("Invalid choice");
                    break;
            }
        }
    }

    public void TheGovernment() {
        while (true) {
            Console.WriteLine("Know your government");
            Console.WriteLine("1: The Governor");
            Console.WriteLine("2: Chief Minister");
            Console.WriteLine("3: Council Of Ministers");
            Console.WriteLine("4: Exit");
            int mla = int.Parse(Console.ReadLine());

            switch (mla) {
                case 1:
                    Console.WriteLine("************The Governor************");
                    Console.WriteLine("Name: Shri. P. S. Sreedharan Pillai");
                    Console.WriteLine("Email: ram@#$@gmail.com");
                    Console.WriteLine("Address: Office of speaker, Goa legislative Assembly\n");
                    break;
                case 2:
                    Console.WriteLine("***********The Chief Minister*******************");
                    Console.WriteLine("Name: Shri. Pramod Sawant");
                    Console.WriteLine("Email: jous@email id");
                    Console.WriteLine("Address: Office of speaker, Goa legislative Assembly\n");
                    break;
                case 3:
                    Council();
                    break;
                case 4:
                    return;
                default:
                    Console.WriteLine("Invalid choice");
                    break;
            }
        }
    }
}

class Assembly : Mla {
    public void MyAssemble() {
        while (true) {
            Console.WriteLine("1: The Legislature");
            Console.WriteLine("2: The Government");
            Console.WriteLine("3: The House");
            Console.WriteLine("4: Contact Us");
            Console.WriteLine("5: Exit");
            int menu = int.Parse(Console.ReadLine());

            switch (menu) {
                case 1:
                    Console.WriteLine("The Legislature:");
                    ListMla();
                    break;
                case 2:
                    Console.WriteLine("The Government:");
                    TheGovernment();
                    break;
                case 3:
                    Console.WriteLine("The House:");
                    ManageBills();
                    break;
                case 4:
                    Console.WriteLine("Contact Us:");
                    Console.WriteLine("Email: info@goa.gov.in\n");
                    break;
                case 5:
                    Console.WriteLine("Exiting...");
                    return;
                default:
                    Console.WriteLine("Invalid choice");
                    break;
            }
        }
    }
}

class Program {
    static void Main(string[] args) {
        Assembly assembly = new Assembly();
        assembly.MyAssemble();
    }
}
